cpg_infra/billing_aggregator/aggregate/utils.py

```diff
@@ -1223,3 +1238,60 @@ def reformat_bigqquery_labels(data: list[dict[str, Any]]) -> dict[str, Any]:
                 labels[k] = v
 
     return labels
+
+
+def retrieve_stored_ids(
+    start: datetime,
+    end: datetime,
+    service_id: str,
+    table: str = GCP_AGGREGATE_DEST_TABLE,
+) -> set[str]:
+    """
+    Retrieve all the stored ids using seqr- and hail- prefixes
+    """
+    logger.info(
+        f'Retrieving stored ids for {table} between {start} and {end}',
+    )
+
+    if '`' in table:
+        raise ValueError('Table name cannot contain backticks')
+
+    if service_id not in ('seqr', 'hail'):
+        raise ValueError(f'Invalid service_id: {service_id}')
+
+    if table not in (GCP_AGGREGATE_DEST_TABLE, GCP_BILLING_BQ_TABLE):
+        raise ValueError(f'Invalid table: {table}')
+
+    _query = f"""
+        SELECT id FROM `{table}`
+        WHERE DATE_TRUNC(usage_end_time, DAY) BETWEEN @window_start AND @window_end
+        AND id LIKE '{service_id}-%';
+    """  # noqa: S608 both tables and service_id are checked for validity
+
+    job_config = bq.QueryJobConfig(
+        query_parameters=[
+            bq.ScalarQueryParameter(
+                'window_start',
+                'STRING',
+                start.strftime('%Y-%m-%d'),
+            ),
+            bq.ScalarQueryParameter(
+                'window_end',
+                'STRING',
+                end.strftime('%Y-%m-%d'),
+            ),
+        ],
+    )
+
+    records = set()
+    try:
+        result = get_bigquery_client().query(_query, job_config=job_config).result()
+        records = set(result.to_dataframe()['id'])
+    except Exception as e:  # noqa: BLE001
+        logger.error(e)
+
+    logger.info(
+        f'Retrieved {len(records)} stored ids',
+    )
+
+    return records
```

Comment:
Why wouldn't we want to FAIL if we can't get the records, and return an empty set - this would mean duplicate records

Reply:
Yes, makes sense, removed the try/except.